File: TrackBudget/TrackBudget.Domain/Entities/Budget.cs
using TrackBudget.Domain.Common;

namespace TrackBudget.Domain.Entities;

public class Budget : BaseEntity
{
    public decimal Limit { get; set; }

    public int Month { get; set; }

    public int Year { get; set; }

    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
}

File: TrackBudget/TrackBudget.Client/src/types/budget.ts
export interface Budget {
    id: string;
    userId: string;
    limit: number;
    spent: number;
    remaining: number;
    month: number;
    year: number;
    categoryId: string;
    categoryName: string;
}

export interface CreateBudget {
    limit: number;
    month: number;
    year: number;
    categoryId: string;
}

export interface UpdateBudget {
    limit: number;
    month: number;
    year: number;
    categoryId: string;
}
